draft-ietf-detnet-dp-sol-mpls-02.xml

```diff
@@ -1113,37 +1004,337 @@ Examples:
+   together with a Generic Associated Channel Label (GAL) <xref
+   target="RFC5586"/> MAY be used in place of a d-CW.
    </t>
    <t>
    Similarly, an Entropy Label Indicator/Entropy Label (ELI/EL)
-   <xref target="RFC6790"/> may be carried below the S-Label in the label stack in order to control or prevent ECMP treatment for packets in a DetNet flow (the use of the same EL for all of the packets in a flow forces the flow to follow the same path when otherwise ECMP is in use in a network). However, as previously stated in <xref target="dn-gen-encap-solution"/>, the use of ECMP in a DetNet domain is not recommended.
-	</t>
+   <xref target="RFC6790"/> MAY be carried below the S-Label in the
+   label stack in networks where DetNet flows would otherwise received ECMP treatment.
+   When ELs are used, the same EL value SHOULD be used for
+   all of the packets sent using a specific S-Label to force the flow to follow the
+   same path.   However, as previously stated in <xref target="dn-gen-encap-solution"/>, the use of ECMP in a DetNet domain is NOT RECOMMENDED.
+   </t>
+   <t>
+     When receiving a DetNet MPLS flow, an implementation MUST identify
+     the app flow associated with the incoming packet based on the
+     S-Label.  When a node is using platform labels for S-Labels, no
+     additional information is needed as the S-label uniquely identifies
+     the app flow.  In the case where platform labels are not used, one
+     or more F-Labels proceeding the S-Label MUST be used together with
+     the S-Label to uniquely identify the incoming app-flows. When PHP
+     is used, the incoming interface MAY also be used to together with
+     any present F-Label(s) and the S-Label to uniquely identify an
+     incoming app-flows.  Note that choice to use platform label space
+     for S-Label or S-Label plus one or more F-Labels to identify app
+     flows is a local implementation choice, with one caveat.  When one
+     or more F-labels, or incoming interface, is needed together with an
+     S-Label to uniquely identify, the controller plane MUST ensure that
+     incoming DetNet MPLS packets arrive with the needed information
+     (F-label(s) and/or incoming interface); the details of such are
+     outside the scope of this document.
+   </t>
    <t>
-    The S-Label for a specific DetNet flow is unique to that DetNet flow on a specific node, but
-    is not required to be identical with the S-Label for that DetNet flow in any other node
-    within the DetNet domain. Thus the S-Label can only be used to identify the DetNet flow
-    at the intended receiving node.
+     While NOT REQUIRED, the use of platform labels for S-Labels matches
+     other pseudowire encapsulation.  This implementation choice also
+     impacts PEF and POF processing as described in the next section.
    </t>
+   <section anchor="pef-requirements" title="Packet Elimination Function

```

Comment:
Why no PRF section? We should highlight that in case of replication the S-label swap can result in different S-label values (i.e., member flows may have different S-label).